File: src/main/java/me/taubsie/dungeonhub/application/exceptions/CommandExecutionException.java
package me.taubsie.dungeonhub.application.exceptions;

import lombok.NoArgsConstructor;
import me.taubsie.dungeonhub.common.exceptions.ProgramException;

import java.io.Serial;

@NoArgsConstructor
public class CommandExecutionException extends ProgramException
{
    @Serial
    private static final long serialVersionUID = 6707538877645992492L;


    public CommandExecutionException(String s) {
        super(s);
    }

    public CommandExecutionException(String message, Throwable cause) {
        super(message, cause);
    }

    public CommandExecutionException(Throwable cause) {
        super(cause);
    }
}

File: src/main/kotlin/me/taubsie/dungeonhub/application/exceptions/CommandExecutionException.kt
package me.taubsie.dungeonhub.application.exceptions

import me.taubsie.dungeonhub.common.exceptions.ProgramException
import java.io.Serial

open class CommandExecutionException : ProgramException {
    constructor() : super()

    constructor(s: String?) : super(s)

    constructor(message: String?, cause: Throwable?) : super(message, cause)

    constructor(cause: Throwable?) : super(cause)

    companion object {
        @Serial
        private val serialVersionUID = 6707538877645992492L
    }
}